src/sdk/helpers/requestsManager.ts

```diff
@@ -0,0 +1,187 @@
+    }
+
+    checkArrayAndFilterNull<T>(array: T[]): T[] {
+        if (!array || !Array.isArray(array)) {
+            throwError('No requests found', SdkErrors.RequestsNotFound);
+        }
+
+        const response = array.filter((request) => request !== null);
+
+        if (response.length === 0) {
+            throwError('No requests found in array', SdkErrors.RequestsNotFound);
+        }
+
+        return response;
+    }
+
+    getRequests(): WalletRequest[] {
+        return this.requests;
+    }
+
+    /**
+     * Verifies the WalletRequests are valid requests signed by valid DIDs
+     * @throws if the requests are not valid
+     */
+    async verify(): Promise<void> {
+        for (const request of this.requests) {
+            if (!(await request.verify())) {
+                if (request instanceof LoginRequest) {
+                    throwError(
+                        `Invalid request for ${request.getType()} ${request.getPayload().origin}`,
+                        SdkErrors.JwtNotValid
+                    );
+                } else if (request instanceof DataSharingRequest) {
+                    throwError(`Invalid request for ${request.getType()} `, SdkErrors.JwtNotValid);
+                }
+            }
+        }
+    }
+
+    checkReferrerOrigin(): void {
+        const docReferrer = document.referrer;
+
+        if (!docReferrer) throwError('No referrer found', SdkErrors.ReferrerEmpty);
+
+        const referrerOrigin = new URL(docReferrer).origin;
+
+        const myRequest = this.getLoginRequestsOrThrow().find(
+            (request) => request.getPayload().origin === referrerOrigin
+        );
+
+        if (!myRequest) {
+            const msg =
+                `No origins from: ${this.getLoginRequestsOrThrow().map((r) => r.getPayload().origin)} ` +
+                `match referrer: ${referrerOrigin}`;
+
+            throwError(msg, SdkErrors.WrongOrigin);
+        }
+    }
+
+    getLoginRequests(): LoginRequest[] | undefined {
+        const response = this.requests.filter((request) => request instanceof LoginRequest);
+
+        return response.length > 0 ? response : undefined;
+    }
+
+    getLoginRequestsOrThrow(): LoginRequest[] {
+        const response = this.getLoginRequests();
+
+        if (!response) {
+            throwError('No LoginRequests found', SdkErrors.RequestsNotFound);
+        }
+
+        return response;
+    }
+
+    getDataSharingRequests(): DataSharingRequest[] | undefined {
+        const response = this.requests.filter((request) => request instanceof DataSharingRequest);
+
+        return response.length > 0 ? response : undefined;
+    }
+
+    getDataSharingRequestsOrThrow(): DataSharingRequest[] {
+        const response = this.getDataSharingRequests();
+
+        if (!response) {
+            throwError('No DataSharingRequests found', SdkErrors.RequestsNotFound);
+        }
+
+        return response;
+    }
+
+    getLoginRequestWithSameOriginOrThrow(): LoginRequest {
```

Comment:
where this is used? i didn't found it

Reply:
https://github.com/Tonomy-Foundation/Tonomy-App-Websites/pull/188/files#diff-37c0cb2c30e6c4ae472b0b670984a9848de8918258585cb14908f75d6810f14fR319